File: src/services/cmds/config-explain.ts
import { invoke } from '@tauri-apps/api/core'

export type ConfigDiffChangeType = 'added' | 'removed' | 'modified'

export interface ConfigDiffChange {
  path: string
  changeType: ConfigDiffChangeType
  beforeType: string | null
  afterType: string | null
}

export interface ConfigSectionSummary {
  path: string
  beforeCount: number | null
  afterCount: number | null
  delta: number | null
}

export interface ConfigDiffReport {
  changed: boolean
  explanation: string
  changes: ConfigDiffChange[]
  sectionSummaries: ConfigSectionSummary[]
}

export async function explainConfigDiff(
  beforeYaml: string,
  afterYaml: string,
) {
  return invoke<ConfigDiffReport>('explain_config_diff', {
    beforeYaml,
    afterYaml,
  })
}

File: src-tauri/src/cmd/config_explain.rs
use super::{CmdResult, StringifyErr};
use crate::core::config_explain::{ConfigDiffReport, explain_config_diff as explain_config_diff_impl};
use smartstring::alias::String;

#[tauri::command]
pub async fn explain_config_diff(before_yaml: String, after_yaml: String) -> CmdResult<ConfigDiffReport> {
    explain_config_diff_impl(&before_yaml, &after_yaml).stringify_err()
}
